### PR TITLE
Searching for mobile add-ons should only display compatible results

### LINKED ISSUE #9893: Searching for mobile add-ons should only display compatible results
Searching for mobile add-ons on AMO will currently display add-ons based on their Fennec compatibility, which is no longer relevant. The new source of truth is the Recommended list, and it's likely to stay that way for a long time.

A couple of solutions that come to mind:
* Have the server return only Recommended results for Android.
* Have the front-end force the Recommended filter for all searches.
* Reset all mobile compatibility data so it matches the current Recommended list.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9893

### CODE REVIEW

File: src/core/api/index.js [hunk 1]
```diff
@@ -285,16 +286,21 @@ export function logOutFromServer({ api }: {| api: ApiState |}) {
...
   api: ApiState,
   filters: {|
     query: string,
     addonType?: string,
   |},
 |};
 
-export function autocomplete({ api, filters }: AutocompleteParams) {
+export function autocomplete({
+  _fixFiltersForClientApp = fixFiltersForClientApp,
```
Comment (eviljeff) [inline src/core/api/index.js:298]: I'm probably missing something obvious, but why are we doing it like this - can't `fixFiltersForClientApp` be used directly without having it as a parameter with an underscore?

**Review by eviljeff — APPROVED** [review]
Just a question on why the parameters have been written in that way.

File: src/core/api/index.js (hunk 1 above)
Comment (willdurand) [reply src/core/api/index.js:298]: That is dependency injection, mainly for testing purposes.

We could use it directly but we wouldn't be able to easily create a mock for this function in a test case.

Comment (eviljeff) [reply src/core/api/index.js:298]: for tests, got it.

**Pull request merged**